CurseForgeBadge.py

```diff
@@ -119,6 +147,38 @@ def mods(project, style='full', before='required for', after='mods', l_colour='E
                         shadow_colour=shadow_colour, logo_colour=logo_colour), 200, {'Content-Type': 'image/svg+xml'}
 
 
+@app.route('/supported/<project>.svg')
+@app.route('/supported/<project>(<l_colour>).svg')
+@app.route('/supported/<style>_<project>.svg')
+@app.route('/supported/<style>_<project>(<l_colour>).svg')
+@app.route('/supported/<style>_<project>_<before>_<after>.svg')
+@app.route('/supported/<style>_<project>_<before>_<after>(<l_colour>).svg')
+@app.route('/supported/<style>_<project>_<before>_<after>(<l_colour>-<r_colour>-<text_colour>-<shadow_colour>-<logo_colour>).svg')
+def supported(project, style='full', before='supported by', after='mods', l_colour='E04E14', r_colour='2D2D2D', text_colour='fff',
+              shadow_colour='010101', logo_colour='1C1C1C'):
+    template = open_template('curseShield.svg', request.args)
+    replacement = before + ' '
+    required = CFReader.get_dependents(project, 'required')
+    optional = CFReader.get_dependents(project, 'optional')
+    total = "{:,}".format(long(required.replace(",", "")) + long(optional.replace(",", "")))
+    if style == 'short':
+        splitted = total.split(',')
+        first_number = splitted[0][0]
+        padding_zeros = '0' * (len(splitted[0]) - 1)
+        post_fix = ('M+' if len(splitted) > 2 else ('k+' if len(splitted) > 1 else ''))
```

Comment:
I wonder if any mod will ever reach these conditions :D (guess it is future proof)